lib/Driver/DarwinToolChains.cpp

```diff
@@ -399,18 +330,62 @@ toolchains::Darwin::constructInvocation(const LinkJobAction &job,
     SmallString<128> StaticRuntimeLibPath;
     getRuntimeLibraryPath(StaticRuntimeLibPath, context.Args, /*Shared=*/false);
     Arguments.push_back(context.Args.MakeArgString(StaticRuntimeLibPath));
+
     Arguments.push_back("-lc++");
     Arguments.push_back("-framework");
     Arguments.push_back("Foundation");
     Arguments.push_back("-force_load_swift_libs");
   } else {
     Arguments.push_back(context.Args.MakeArgString(RuntimeLibPath));
-    // FIXME: We probably shouldn't be adding an rpath here unless we know ahead
-    // of time the standard library won't be copied. SR-1967
-    Arguments.push_back("-rpath");
-    Arguments.push_back(context.Args.MakeArgString(RuntimeLibPath));
+
+    if (context.Args.hasArg(options::OPT_toolchain_stdlib_rpath)) {
+      // If the user has explicitly asked for a toolchain stdlib, we should
+      // provide one using -rpath. This used to be the default behaviour but it
+      // was considered annoying in at least the swiftpm scenario (see
+      // https://bugs.swift.org/browse/SR-1967) and is obsolete in all
+      // scenarios of deploying for swift-in-the-OS. We keep it here as an
+      // optional behaviour so that people downloading snapshot toolchains for
+      // testing new stdlibs will be able to link to the stdlib bundled in that
+      // toolchain.
```

Comment:
We worked out all these careful details about how to deal with Swift in downloadable toolchains differently from Swift in Xcode. Do we have to do any tweaks to the downloadable toolchain configuration? @bob-wilson @Rostepher @shahmishal

Reply:
I think that is all in place now, e.g., https://github.com/apple/swift/pull/20578

Reply:
Looking at the context, this is a little different. We don't expect this option to be used on Apple platforms with Swift-in-the-OS, but keeping the support in the compiler seems harmless.